microhapulator/workflows/analysis.smk

```diff
@@ -117,16 +116,13 @@ rule map_sort_and_index:
     output:
         bam="analysis/{sample}/{sample}.bam",
         bai="analysis/{sample}/{sample}.bam.bai",
-        counts="analysis/{sample}/{sample}-mapped-reads.txt",
+        stats="analysis/{sample}/{sample}.bam.stats",
     threads: 32
     shell:
         """
         bwa mem -t {threads} {input.fasta} {input.fastq} | samtools view -b | samtools sort -o {output.bam}
         samtools index {output.bam}
-        echo -n "Total reads: " > {output.counts}
-        samtools view -c -F 2304 {output.bam} >> {output.counts}
-        echo -n "Mapped reads: " >> {output.counts}
-        samtools view -c -F 2308 {output.bam} >> {output.counts}
+        samtools stats {output.bam} > {output.stats}
```

Comment:
Using `samtools stats` instead of `samtools view` to count total reads and mapped reads.